cmd/katib-controller/v1beta1/main.go

```diff
@@ -131,27 +133,40 @@ func main() {
 		CertDir: consts.CertDir,
 	})
 
+	if err := mgr.AddReadyzCheck("webhook", hookServer.StartedChecker()); err != nil {
+		log.Error(err, "Unable to add readyz check cert")
+		os.Exit(1)
+	}
+
 	ctx := signals.SetupSignalHandler()
 	certsReady := make(chan struct{})
-	defer close(certsReady)
 
 	// The setupControllers will register controllers to the manager
 	// after generated certs for the admission webhooks.
 	go setupControllers(mgr, certsReady, hookServer)
 
 	if initConfig.CertGeneratorConfig.Enable {
+		err := mgr.AddReadyzCheck("cert", func(_ *http.Request) error {
```

Comment:
If not specified, readinessProbe checks all readiness checkers. the name refers to the name of the checker, not [readinessEndpointName](https://github.com/kubernetes-sigs/controller-runtime/blob/1ed345090869edc4bd94fe220386cb7fa5df745f/pkg/manager/internal.go#L92).

https://github.com/kubernetes-sigs/controller-runtime/pull/1100

https://github.com/kubernetes-sigs/controller-runtime/blob/1ed345090869edc4bd94fe220386cb7fa5df745f/pkg/healthz/healthz.go#L147-L159

Reply:
@forsaken628 @tenzen-y Just for user visibility, maybe we should explicitly call out them in the controller Deployment ? It might be hard for user to check code to understand all probs.

Reply:
I think the vast majority of case users only care about aggregation probes. We could add a comment somewhere.

Reply:
I just want to keep it consistent between Training Operator: https://github.com/kubeflow/training-operator/blob/master/manifests/base/deployment.yaml#L53 and Katib controller.

Reply:
@tenzen-y Any thoughts ?

Reply:
I don't think we need to modify `deployment.yaml`. The individual check is not a new feature for `controller-runtime`, and usually users don't need to know it exists.